torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -91,13 +91,23 @@ ProcessGroupNCCL::WorkNCCL::WorkNCCL(const std::vector<at::Device>& devices)
 
 ProcessGroupNCCL::WorkNCCL::~WorkNCCL() {}
 
-// Check if the NCCL kernels are queued on the GPUs
 bool ProcessGroupNCCL::WorkNCCL::isCompleted() {
+  return finishedGPUExecution();
+}
+
+bool ProcessGroupNCCL::WorkNCCL::isSuccess() const {
   return true;
 }
 
+std::exception_ptr ProcessGroupNCCL::WorkNCCL::exception() const {
+  throw std::runtime_error(
+      "exception() is not supported by NCCL process "
+      "group's work, since isSuccess() will always return true, and "
+      "isCompleted() and wait() will either succeed or throw");
+}
+
 // Helper that checks if the NCCL kernels are completed on the GPUs
-bool ProcessGroupNCCL::WorkNCCL::finishedGPUExecution() const {
+bool ProcessGroupNCCL::WorkNCCL::finishedGPUExecution() {
```

Comment:
why removing this const?

Reply:
It is now called from the `isCompleted()` function (it was not used at all before it looks like), and `isCompleted()` is not const (it may poll for completion and set the completed flag and optionally the exception ptr).